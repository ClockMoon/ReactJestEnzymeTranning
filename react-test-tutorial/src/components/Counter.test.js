import React from "react"
import renderer from "react-test-renderer"
import Counter from "./Counter"

describe("Counter 테스트", () => {
  let component = null

  it("렌더링 테스트", () => {
    component = renderer.create(<Counter />)
  })

  it("스냅샷과 비교", () => {
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("더하기 기능", () => {
    component.getInstance().onIncrease()
    expect(component.getInstance().state.value).toBe(2)
    const tree = component.toJSON() // re-render
    expect(tree).toMatchSnapshot() // 스냅샷 비교
  })

  it("빼기 기능", () => {
    component.getInstance().onDecrease()
    expect(component.getInstance().state.value).toBe(1) // value 값이 1인지 확인
    const tree = component.toJSON() // re-render
    expect(tree).toMatchSnapshot() // 스냅샷 비교
  })
})
