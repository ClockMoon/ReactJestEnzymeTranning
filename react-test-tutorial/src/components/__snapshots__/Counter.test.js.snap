// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter 테스트 더하기 기능 1`] = `
<div>
  <h1>
    카운터!
  </h1>
  <h2>
    2
  </h2>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
</div>
`;

exports[`Counter 테스트 빼기 기능 1`] = `
<div>
  <h1>
    카운터!
  </h1>
  <h2>
    1
  </h2>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
</div>
`;

exports[`Counter 테스트 스냅샷과 비교 1`] = `
<div>
  <h1>
    카운터!
  </h1>
  <h2>
    1
  </h2>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
</div>
`;
